applications/nrf_desktop/src/modules/usb_state.c

```diff
@@ -1227,11 +1259,16 @@ static bool app_event_handler(const struct app_event_header *aeh)
 			__ASSERT_NO_MSG(!initialized);
 			initialized = true;
 
-			if (usb_init()) {
-				LOG_ERR("USB stack initialization failed");
-				module_set_state(MODULE_STATE_ERROR);
+			if (IS_ENABLED(CONFIG_DESKTOP_USB_STATE_INIT_THREAD)) {
+				k_thread_create(&usb_init_thread,
+						usb_init_thread_stack,
+						K_THREAD_STACK_SIZEOF(usb_init_thread_stack),
+						usb_init_thread_fn,
+						NULL, NULL, NULL,
+						CONFIG_DESKTOP_USB_STATE_INIT_THREAD_PRIORITY, 0,
+						K_MSEC(CONFIG_DESKTOP_USB_STATE_INIT_THREAD_DELAY_MS));
 			} else {
-				module_set_state(MODULE_STATE_READY);
+				usb_state_module_init();
```

Comment:
btw. Maybe you could directly call `usb_init_thread_fn` to simplify implementation a bit and show intention here?